src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -413,9 +416,23 @@
       <RazorSonarProjectSpecificOutDir>$(ProjectSpecificOutDir).Razor</RazorSonarProjectSpecificOutDir>
       <RazorSonarProjectInfo>$(RazorSonarProjectSpecificOutDir)\ProjectInfo.xml</RazorSonarProjectInfo>
     </PropertyGroup>
+    <ItemGroup>
+      <RazorCompilationOutFiles Include="$(ProjectSpecificOutDir)\**\*.*"/>
+      <SonarTempFiles Include="$(SonarTemporaryProjectSpecificOutDir)\**\*.*"/>
+    </ItemGroup>
+
     <!-- Move main and Razor results to their final location. -->
-    <MoveDirectory SourceDirectory="$(ProjectSpecificOutDir)" DestinationDirectory="$(RazorSonarProjectSpecificOutDir)" />
-    <MoveDirectory SourceDirectory="$(SonarTemporaryProjectSpecificOutDir)" DestinationDirectory="$(ProjectSpecificOutDir)" />
+    <Move
+      SourceFiles="@(RazorCompilationOutFiles)"
+      DestinationFolder="$(RazorSonarProjectSpecificOutDir)\%(RecursiveDir)" />
```

Comment:
SonarFinishRazorProjectCodeAnalysis is already tested in `Razor_RazorSpecificOutputAndProjectInfo_PreserveUserDefinedErrorLogValue` and `Razor_RazorSpecificOutputAndProjectInfo_AreCopiedToCorrectFolders`